### PR TITLE
Cover/cmds/core/truncate: Add tests.

### PR DESCRIPTION
**Pull request by llogen**
Refactored truncate.go and added tests in truncate_test.go.

Signed-off-by: llogen <christoph.lange@9elements.com>

### CODE REVIEW

File: cmds/core/truncate/truncate.go [hunk 1]
```diff
@@ -34,50 +36,38 @@ var (
...
 
-	for _, fname := range flag.Args() {
+	for _, fname := range args {
 
 		var final int64
 		st, err := os.Stat(fname)
 		if os.IsNotExist(err) && !*create {
 			if err = os.WriteFile(fname, []byte{}, 0o644); err != nil {
-				log.Fatalf("truncate: ERROR: %v\n", err)
+				return fmt.Errorf("truncate: ERROR: %v", err)
```
Comment (rminnich) [inline cmds/core/truncate/truncate.go:62]: I think you might take out the ERROR: part of these error messages? 

Also in the standard command create is true.
Check out this bizarre flag:
       -c, --no-create
              do not create any files

It gets better: 
rminnich@t510:~$ truncate -c -s  0 /tmp/asdfadfadsffd
rminnich@t510:~$ rm !$
rm /tmp/asdfadfadsffd
rm: cannot remove '/tmp/asdfadfadsffd': No such file or directory
rminnich@t510:~$

notice: no file created, no error message. Awesome! What should *we* do? At least, I think, make the flag true.

File: cmds/core/truncate/truncate.go [hunk 2]
```diff
@@ -34,50 +36,38 @@ var (
...
 			}
 			if st, err = os.Stat(fname); err != nil {
-				log.Fatalf("truncate: ERROR: could not stat newly created file: %v\n", err)
+				return fmt.Errorf("truncate: ERROR: could not stat newly created file: %v", err)
```
Comment (rminnich) [inline cmds/core/truncate/truncate.go:65]: in the standard command, this is not an error (!!!)

File: cmds/core/truncate/truncate.go [hunk 3]
```diff
@@ -34,50 +36,38 @@ var (
...
 			}
 		}
 		if *rfile != "" {
 			if st, err = os.Stat(*rfile); err != nil {
-				log.Fatalf("truncate: ERROR: could not stat reference file: %v\n", err)
+				return fmt.Errorf("truncate: ERROR: could not stat reference file: %v", err)
```
Comment (rminnich) [inline cmds/core/truncate/truncate.go:70]:
```
rminnich@t510:~$ truncate -r  /tmp/asdfadfadsffd /tmppppp
truncate: cannot stat '/tmp/asdfadfadsffd': No such file or directory
rminnich@t510:~$
```

It is probably time to start wrapping our errors anyway: 
fmt.Errorf("truncate: %w", err)

File: cmds/core/truncate/truncate.go (hunk 1 above)
Comment (llogen) [reply cmds/core/truncate/truncate.go:62]: Regarding this https://man7.org/linux/man-pages/man1/truncate.1.html the flag create is correct as it is. If you set the -c flag than no file will be created.

File: cmds/core/truncate/truncate.go (hunk 2 above)
Comment (llogen) [reply cmds/core/truncate/truncate.go:65]: The original line was: log.Fatalf("truncate: ERROR: could not stat newly created file: %v\n", err).
This ends up in an os.Exit. So the idea was to just reach an err right into a log.Fatal(err). Like that i don't change the logic of the code.

**Review by rminnich — APPROVED** [review]
I'm still not comfortable with the error handling but let's face it the flag package makes it hard to do the right thing :-)

**Comment by llogen** [comment]
> I'm still not comfortable with the error handling but let's face it the flag package makes it hard to do the right thing :-)
Jep i was also confused and logical it does not make sense at all.

**Pull request merged**